http/action/src/test/java/io/knotx/databridge/http/action/HttpActionTest.java

```diff
@@ -406,155 +391,106 @@ void testSuccessTransitionExpectedAndResponseAsRawText(String contentType, boole
-        new HttpActionOptions()
-            .setEndpointOptions(endpointOptions)
-            .setResponseOptions(responseOptions),
-        ACTION_ALIAS);
-
-    verifyFailingExecution(tested, clientRequest, FRAGMENT, error -> {
-      assertTrue(error instanceof CompositeException);
-      CompositeException exception = (CompositeException) error;
-      assertEquals(1, exception.getExceptions().size());
-      assertTrue(exception.getExceptions().get(0) instanceof ReplyException);
+    verifyExecution(tested, clientRequest, createFragment(), fragmentResult -> {
+      assertNotNull(fragmentResult);
+      assertEquals(ERROR_TRANSITION, fragmentResult.getTransition());
+      assertEquals(EMPTY_JSON, fragmentResult.getFragment().getPayload());
+      JsonObject logs = fragmentResult.getNodeLog().getJsonObject("logs");
+      assertEquals(CompositeException.class.getCanonicalName(),
+          logs.getJsonObject("error").getString("className"));
     }, testContext);
   }
 
   @Test
   @DisplayName("Expect IllegalArgumentException when not existing predicate provided")
-  void expectErrorWhenNotExistingPredicateProvided(VertxTestContext testContext, Vertx vertx) throws Throwable {
+  void expectErrorWhenNotExistingPredicateProvided(VertxTestContext testContext, Vertx vertx)
+      throws Throwable {
     String endpointPath = "/api/exception-wrong-predicate";
-    Set<String> predicates = new HashSet<>();
-    predicates.add(NOT_EXISTING_PREDICATE);
-
-    wireMockServer.stubFor(get(urlEqualTo(endpointPath))
-        .willReturn(aResponse().withBody(JSON_BODY)
-            .withHeader("Content-Type", APPLICATION_JSON)));
 
     ClientRequest clientRequest = prepareClientRequest(MultiMap.caseInsensitiveMultiMap(),
         MultiMap.caseInsensitiveMultiMap(), endpointPath);
+    HttpAction tested = setupTestingInstances(vertx, endpointPath, JSON_BODY, APPLICATION_JSON,
+        NOT_EXISTING_PREDICATE, false, actionLogLevel);
 
-    EndpointOptions endpointOptions = new EndpointOptions()
-        .setPath(endpointPath)
-        .setDomain("localhost")
-        .setPort(wireMockServer.port())
-        .setAllowedRequestHeaderPatterns(Collections.singletonList(Pattern.compile(".*")));
+    verifyExecution(tested, clientRequest, createFragment(), fragmentResult -> {
+      assertNotNull(fragmentResult);
+      assertEquals(ERROR_TRANSITION, fragmentResult.getTransition());
+      assertEquals(EMPTY_JSON, fragmentResult.getFragment().getPayload());
+      JsonObject logs = fragmentResult.getNodeLog().getJsonObject("logs");
+      assertEquals(CompositeException.class.getCanonicalName(),
+          logs.getJsonObject("error").getString("className"));
+    }, testContext);
+  }
 
-    ResponseOptions responseOptions = new ResponseOptions()
-        .setPredicates(predicates)
-        .setForceJson(false);
+  @Test
+  @DisplayName("Expect _error transition, empty payload and DecodeException when service responds with invalid json when json expected")
```

Comment:
Can we add one more case:
given: 
- predicates: ResponsePredicate.SC_SUCCESS, ResponsePredicate.JSON
- response: 500

(web client throws an exception)

then:
- _error transition
- "errors" in node log with appropriate exception

Reply:
@marcinus one more.